dfu: img_util: move flash defines to SoC

nashif opened the pull request:
mcuboot_constraints.h had FLASH information related to the SoC that
should be maintained as part of the SoC and not in the subsystem. Also
fixed Makefiles to check for IMG_UTIL Kconfig and not MCUBOOT.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
dfu: img_util: move flash defines to SoC

mcuboot_constraints.h had FLASH information related to the SoC that
should be maintained as part of the SoC and not in the subsystem. Also
fixed Makefiles to check for IMG_UTIL Kconfig and not MCUBOOT.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

carlescufi commented:
@nvlsianpu please review

carlescufi reviewed (dismissed):
LGTM

nvlsianpu commented on arch/arm/soc/nordic_nrf5/nrf52/soc.h:
```diff
@@ -21,7 +21,13 @@
+/* For IMG_MANAGER  */
+#if defined(CONFIG_SOC_FLASH_NRF5)
+#define FLASH_DRIVER_NAME		CONFIG_SOC_FLASH_NRF5_DEV_NAME
+#define FLASH_ALIGN			4
```
So far for nRF52840 DTS item [write-block-size](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nordic/nrf52840.dtsi#L26) was added. I recommend to add similar path for all of nrf5 devices and use it here.

nvlsianpu replied:
I prepared two commits for update nrf52832 and nrf51822 DT. You can take them from my fork: https://github.com/nvlsianpu/zephyr/tree/flash_alignment_nrf5x

nashif replied:
not sure what you want me to do, can you apply this after we merge this PR?

nvlsianpu replied:
I would just replace this by, and add last two commits form my branch:
`#define FLASH_ALIGN	 FLASH_WRITE_BLOCK_SIZE`

In case you merege this as is (LGTM) I will apply a patch later.

nashif replied:
ok, done

nvlsianpu committed:
dts: nrf: Add flash controller and alignment description for nRF52832

Similar addition was don previous for nRF52840.
Added flash-controller description and moved flash description to
it. Added property for description of
the flash alignment required by write operations.


Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
dts: nrf: Add flash controller and alignment description for nRF51822

Similar addition was don previous for nRF52840.
Added flash-controller description and moved flash description to
it. Added property for description of
the flash alignment required by write operations.


Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
